crypto/crypto_test.go

```diff
@@ -0,0 +1,286 @@
+				return
+			}
+			info := models.GetEncryptionKey(tt.keyName, tt.version)
+			if tt.keyExists {
+				if info == nil {
+					t.Errorf("Key %s not found in AvailableKeys", tt.keyName)
+					return
+				}
+			} else {
+				if info != nil {
+					t.Errorf("Key %s found in AvailableKeys", tt.keyName)
+				}
+				return
+			}
+
+			if info.GetName() != tt.keyName {
+				t.Errorf("GetName() = %v, want %v", info.GetName(), tt.keyName)
+			}
+			if info.GetVersion() != tt.version {
+				t.Errorf("GetVersion() = %v, want %v", info.GetVersion(), tt.version)
+			}
+
+			if tt.privKeyNull {
+				if info.GetPrivKey() != nil {
+					t.Errorf("GetPrivKey() = %v, want nil", info.GetPrivKey())
+				}
+			} else {
+				if info.GetPrivKey() == nil {
+					t.Errorf("GetPrivKey() = %v, want not nil", info.GetPrivKey())
+				}
+			}
+
+			if tt.pubKeyNull {
+				if info.GetPubKey() != nil {
+					t.Errorf("GetPubKey() = %v, want nil", info.GetPubKey())
+				}
+			} else {
+				if info.GetPubKey() == nil {
+					t.Errorf("GetPubKey() = %v, want not nil", info.GetPubKey())
+				}
+			}
+
+		})
+	}
+}
+
+func TestGetEncryptionKey(t *testing.T) {
+	teardownSuite := setupSuite(t)
+	LoadKeysFromEnv()
+	defer teardownSuite(t)
+	type args struct {
+		keyName    string
+		keyVersion string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *models.KeyInfo
+	}{
+		{
+			name: "TestGetEncryptionKey - No errors",
+			args: args{
+				keyName:    "TESTKEY",
+				keyVersion: "1",
+			},
+			want: &models.KeyInfo{
+				PrivKey:   nil,
+				PubKey:    nil,
+				Name:      "TESTKEY",
+				Version:   "1",
+				CreatedAt: time.Time{},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := models.GetEncryptionKey(tt.args.keyName, tt.args.keyVersion); got != nil {
+				if got.GetName() != tt.want.GetName() {
+					t.Errorf("GetName() = %v, want %v", got.GetName(), tt.want.GetName())
+				}
+
+				if got.GetVersion() != tt.want.GetVersion() {
+					t.Errorf("GetVersion() = %v, want %v", got.GetVersion(), tt.want.GetVersion())
+				}
+
+				if got.GetPrivKey() == nil {
+					t.Errorf("GetPrivKey() = %v, want not nil", got.GetPrivKey())
+				}
+
+				if got.GetPubKey() == nil {
+					t.Errorf("GetPubKey() = %v, want not nil", got.GetPubKey())
+				}
+			}
+		})
+	}
+}
+
+func TestEncryptRSA(t *testing.T) {
+	teardownSuite := setupSuite(t)
+	LoadKeysFromEnv()
+	defer teardownSuite(t)
+	type args struct {
+		keyName string
+		data    []byte
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *models.Payload
+		wantErr bool
+	}{
+		{
+			name: "TestEncryptRSA - No errors",
+			args: args{
+				keyName: "TESTKEY",
+				data:    []byte("data"),
+			},
+			want: &models.Payload{
+				Data:          "data",
+				EncryptedData: "9zE+AFpfb3PhIfdaOlPxXZAVHb3oEiTxMYcIoDuaYVs=",
+			},
+			wantErr: false,
+		},
+		{
+			name: "TestEncryptRSA - No errors",
+			args: args{
+				keyName: "TESTKEYR",
```

Comment:
TESTKEYR doesn't exists

Reply:
if key doesn't exist. not returning error, but returning the same data as input